server/src/main/java/org/apache/druid/client/HttpServerInventoryView.java

```diff
@@ -204,15 +214,26 @@ private DruidServer toDruidServer(DiscoveryDruidNode node)
             }
         );
 
-        scheduleSyncMonitoring();
+        ScheduledExecutors.scheduleAtFixedRate(
+            executor,
+            Duration.standardSeconds(60),
+            Duration.standardMinutes(5),
+            this::checkAndResetUnhealthyServers
+        );
+        ScheduledExecutors.scheduleAtFixedRate(
+            executor,
+            Duration.standardSeconds(30),
+            Duration.standardMinutes(1),
+            this::emitServerStatusMetrics
```

Comment:
I just realized that one benefit of having a separate executor is that metrics get emitted even if thread pool is busy for other reasons. Thought could be done later.

Reply:
Yeah, true. If I am making other changes, I will move metric emission and the check and reset logic to the separate executor.